src/plugin.js

```diff
@@ -1,85 +1,64 @@
 'use strict';
-var Joi = require('joi');
+const Joi = require('joi');
+const _ = require('lodash');
 
-exports.register = function(plugin, options, next){
-  var schemaOptions =  Joi.alternatives().try(
-    Joi.object().keys({
-      lower_case: Joi.boolean().required(),
-      camel_case: Joi.boolean().when('lower_case', { is: true, then: Joi.invalid(true) }),
-      all_caps: Joi.boolean().when('lower_case', { is: true, then: Joi.invalid(true) }),
-      first_cap: Joi.boolean().when('lower_case', { is: true, then: Joi.invalid(true) }),
-      delete_original: Joi.boolean().optional()
-    }),
-    Joi.object().keys({
-      lower_case: Joi.boolean().when('camel_case', { is: true, then: Joi.invalid(true) }),
-      camel_case: Joi.boolean().required(),
-      all_caps: Joi.boolean().when('camel_case', { is: true, then: Joi.invalid(true) }),
-      first_cap: Joi.boolean().when('camel_case', { is: true, then: Joi.invalid(true) }),
-      delete_original: Joi.boolean().optional()
-    }),
-    Joi.object().keys({
-      lower_case: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      camel_case: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      all_caps: Joi.boolean().required(),
-      first_cap: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      delete_original: Joi.boolean().optional()
-    }),
-    Joi.object().keys({
-      lower_case: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      camel_case: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      all_caps: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      first_cap: Joi.boolean().required(),
-      delete_original: Joi.boolean().optional()
-    })
-  );
+const schemaOptions = 
+  Joi.object().keys({
+    lowerCase: Joi.boolean(),
+    schemaCase: Joi.object(),
+    deleteOriginal: Joi.boolean().optional()
+  }).xor('lowerCase', 'schemaCase');;
 
-  var validate = Joi.validate(options, schemaOptions);
+function generateKeyMapping(schema) {
+  const providedSchema = _.mapKeys(schema, _.rearg(_.identity, 1));
+  const keyMapping = _.mapKeys(_.mapValues(providedSchema, function(value, key) { return key; }), function(value) { return _.toLower(value);});
```

Comment:
Do this is two steps. Combining the logic here just makes this confusing, and in practice you're doing two loops anyways. Either create two variables or use https://lodash.com/docs/4.17.11#chain to chain the actions in two distinct steps